agent/agent.go

```diff
@@ -114,28 +114,24 @@ func (a *Agent) gatherer(
 	defer ticker.Stop()
 
 	for {
-		var outerr error
-
-		acc := NewAccumulator(input.Config, metricC)
-		acc.SetDebug(a.Config.Agent.Debug)
-		acc.SetPrecision(a.Config.Agent.Precision.Duration,
-			a.Config.Agent.Interval.Duration)
-		acc.setDefaultTags(a.Config.Tags)
-
-		internal.RandomSleep(a.Config.Agent.CollectionJitter.Duration, shutdown)
+		go func() {
+			acc := NewAccumulator(input.Config, metricC)
+			acc.SetDebug(a.Config.Agent.Debug)
+			acc.SetPrecision(a.Config.Agent.Precision.Duration,
+				a.Config.Agent.Interval.Duration)
+			acc.setDefaultTags(a.Config.Tags)
 
-		start := time.Now()
-		gatherWithTimeout(shutdown, input, acc, interval)
-		elapsed := time.Since(start)
+			internal.RandomSleep(a.Config.Agent.CollectionJitter.Duration, shutdown)
 
-		if outerr != nil {
-			return outerr
-		}
-		if a.Config.Agent.Debug {
-			log.Printf("Input [%s] gathered metrics, (%s interval) in %s\n",
-				input.Name, interval, elapsed)
-		}
+			start := time.Now()
+			gatherWithTimeout(shutdown, input, acc, interval)
+			elapsed := time.Since(start)
 
+			if a.Config.Agent.Debug {
+				log.Printf("Input [%s] gathered metrics, (%s interval) in %s\n",
+					input.Name, interval, elapsed)
+			}
+		}()
```

Comment:
This change can result in kicking off a new `Gather()` operation on a plugin before the previous one has completed (if `ticker.C` fires). This seems very dangerous as the plugins aren't coded to handle this. They're going to stomp all over persisted variables (simultaneous access).